graphics/blender/Portfile

```diff
@@ -281,4 +340,24 @@ variant manpage description {Build manpage} {
     configure.args-append   -DWITH_DOC_MANPAGE=ON
 }
 
+variant usd_monolithic \
+    description {Build Blender using USD's monolithic library option, instead of modular USD libaries. Use this variant if you intend to install the USD port using the +monolithic variant.} \
+{
+    require_active_variants usd monolithic
+}
+
+if {![variant_isset usd_monolithic]} {
+    require_active_variants usd {} monolithic
+}
+
+variant embree_static \
+    description {Build Blender using static Embree libraries, instead of a dynamic library. Use this variant if you intend to install the Embree port using the +static variant.} \
+{
+    require_active_variants embree static
+}
+
+if {![variant_isset embree_static]} {
+    require_active_variants embree {} static
+}
```

Comment:
Why is all of this happening? It is brittle / confusing to offer variants that change what libraries are installed and to require matched sets of variants across different ports. Why should there be a monolithic variant of usd or a static variant of embree at all?

Reply:
> It is brittle / confusing to offer variants that change what libraries are installed and to require matched sets of variants across different ports.

The variants don't actually change what libraries are installed. All they do is make sure that if the user installs the variants of those libraries, that Blender has to be built in a certain way. I don't think that using variants is any more brittle than simply checking for which variant of the library is installed before the build phase begins, because any mismatched situations will give users feedback as to how they can get to a working, non-broken state. If you feel that doing a check before the build phase is sufficient for the situation where a user goes in and installs one of the libraries' variants after building Blender, then fine, but then please don't blame me later if users complain that they aren't getting any sort of messages that Blender is broken after installing the library variants.

> Why should there be a monolithic variant of usd or a static variant of embree at all?

These are the main modes of use for usd and embree, and other software projects which depend on usd or embree may need to use those variants. For example, there are portions of Embree that only get compiled when compiling Embree as a static library. Specifically, these portions pertain to Intel instruction set extensions such as AVX2, SSE2, SSE4, etc. When Embree is compiled as a dynamic library, it's not necessary to compile modules for these instruction sets separately. However, if Embree is compiled as a static library, modules for each of the instruction sets needs to be compiled, which can result in a very large static library. (See https://github.com/macports/macports-ports/pull/9366#issuecomment-742020416)